WalletWasabi.Gui/Global.cs

```diff
@@ -285,10 +285,10 @@ private void RegisterFeeRateProviders()
 			HostedServices.Register<HybridFeeProvider>(new HybridFeeProvider(HostedServices.Get<ThirdPartyFeeProvider>(), HostedServices.GetOrDefault<RpcFeeProvider>()), "Hybrid Fee Provider");
 		}
 
-		private void RegisterCoinJoinComponents()
+		private void RegisterCoinJoinComponents(SystemAwakeChecker? systemAwakeChecker)
 		{
 			HostedServices.Register<RoundStateUpdater>(new RoundStateUpdater(TimeSpan.FromSeconds(5), new WabiSabiHttpApiClient(BackendHttpClientFactory.NewBackendHttpClient(Mode.SingleCircuitPerLifetime))), "Round infor updater");
-			HostedServices.Register<CoinJoinManager>(new CoinJoinManager(WalletManager, HostedServices.Get<RoundStateUpdater>(), BackendHttpClientFactory, Config.ServiceConfiguration), "CoinJoin Manager");
+			HostedServices.Register<CoinJoinManager>(new CoinJoinManager(WalletManager, systemAwakeChecker, HostedServices.Get<RoundStateUpdater>(), BackendHttpClientFactory, Config.ServiceConfiguration), "CoinJoin Manager");
```

Comment:
I don't understand why the `CoinJoinManager` needs to know about the `SystemAwakeChecker`, it should not. The ``SystemAwakeChecker` has access to all what it needs to decide what to do without having the `CoinJoinManager` to do anything.

Reply:
ACK!